File: 2021-06/0614猜数字大小/main.cpp
/** 
 * Forward declaration of guess API.
 * @param  num   your guess
 * @return 	     -1 if num is lower than the guess number
 *			      1 if num is higher than the guess number
 *               otherwise return 0
 * int guess(int num);
 */



class Solution {
public:
    int guessNumber(int n) {
        int left = 1;
        int right = n;
        int mid = (right - left) / 2 + left; 
        while(guess(mid)!=0){
            if(guess(mid) == 1){
                left = mid + 1;
            }else{
                right = mid - 1;
            }
            mid = (right - left) / 2 + left;
        }
        return mid;
    }
};

File: 2021-06/0614猜数字大小/main.go
package main

/**
 * Forward declaration of guess API.
 * @param  num   your guess
 * @return 	     -1 if num is lower than the guess number
 *			      1 if num is higher than the guess number
 *               otherwise return 0
 * func guess(num int) int;
 */

func guess(guess int, target int) int {
	if target == guess {
		return 0
	} else if guess > target {
		return -1
	} else {
		return 1
	}
}

func guessNumber(n int, target int) int {

	left := 1
	right := n
	for left < right {
		mid := (right-left)/2 + left
		if guess(mid, target) == 0 {
			return mid
		} else if guess(mid, target) == -1 {
			right = mid
		} else {
			left = mid + 1
		}
	}
	return left

}
